tracer/src/Datadog.Trace/AppSec/Transports/Http/HttpTransport.Core.cs

```diff
@@ -40,6 +51,48 @@ public IHeadersCollection GetResponseHeaders()
         {
             return new HeadersCollectionAdapter(_context.Response.Headers);
         }
+
+        public void WriteBlockedResponse(string templateJson, string templateHtml)
+        {
+            _context.Items["block"] = true;
+            var httpResponse = _context.Response;
+            httpResponse.Clear();
+            httpResponse.Headers.Clear();
+            httpResponse.StatusCode = 403;
+            var syncIOFeature = _context.Features.Get<IHttpBodyControlFeature>();
+            if (syncIOFeature != null)
+            {
+                // allow synchronous operatoin for net core >=3.1 otherwise invalidoperation exception
+                syncIOFeature.AllowSynchronousIO = true;
```

Comment:
This is kinda horrible, is there no way we can make this async up the stack? 🙁 Last thing we really want is thread starvation and hangs 😬